src/modules/AddOnProduct.rb

```diff
@@ -2209,6 +2224,128 @@ def add_rename(old_name, new_name)
+    end
+
+    # Determine whether a product rename is present on a given hash
+    #
+    # @param renames  [Hash] Product renames in form old_name => [new_name1, new_name2, ...]
+    # @param old_name [String] Old product's name
+    # @param new_name [String] New product's name
+    # @return [Boolean] +true+ if the product was renamed; otherwise, +false+.
+    def renamed_at?(renames, old_name, new_name)
+      return false unless renames[old_name]
+      renames[old_name].include?(new_name)
+    end
+
+    # Get product renames from libzypp
+    #
+    # @see names_from_product_package
+    # @see add_rename_to_hash
+    def product_renames_from_libzypp
+      renames = {}
+      products = Pkg.ResolvableProperties("", :product, "") # Dependencies are not included in this call
+      products.each do |product|
+        renames = names_from_product_package(product["product_package"]).reduce(renames) do |hash, rename|
+          add_rename_to_hash(hash, rename, product["name"])
+        end
+      end
+      renames
+    end
+
+    # Regular expresion to extract the product name
+    # @see product_name_from_dep
+    PRODUCTS_REGEXP = /\Aproduct(?::|\()([^\)\b]+)/
+
+    # Extracts product's name from dependency
+    #
+    # It supports two different formats: "product:NAME" and "product(NAME)"
+    #
+    # @see renamed_by_libzypp?
+    def product_name_from_dep(dependency)
+      matches = PRODUCTS_REGEXP.match(dependency)
+      matches.nil? ? nil : matches[1]
+    end
+
+    # Add a product's rename to a given hash
+    #
+    # It will be ignored if:
+    #
+    # * old and new names are the same or
+    # * the rename it's already included
+    #
+    # @param renames  [Hash]   Renames following the format <old_name> => [ <new_name> ]
+    # @param old_name [String] Old product's name
+    # @param new_name [String] New product's name
+    # @return [Hash] Product renames hash adding the new rename if needed
+    #
+    # @see #add_rename
+    def add_rename_to_hash(renames, old_name, new_name)
+      return renames if old_name == new_name || renamed_at?(renames, old_name, new_name)
+      log.info "Adding product rename: '#{old_name}' => '#{new_name}'"
+      renames.merge(old_name => new_name) do |key, old_val, new_val|
+        old_val.nil? ? [new_val] : old_val + [new_val]
+      end
+    end
+
+    # Determines the products for a product package
+    #
+    # The renames are extracted from the "obsoletes" and "provides" dependencies
+    # (renames like 'SUSE_SLES' to 'SLES' are defined in the "provides"
+    # dependencies only).
+    #
+    # @return [Array<String>] Old names
+    def names_from_product_package(package_name)
+      # Get package dependencies (not retrieved when using Pkg.ResolvableProperties)
+      packages = Pkg.ResolvableDependencies(package_name, :package, "")
+      return [] if packages.nil? || packages.empty? || !packages.first.key?("deps")
```

Comment:
I think it would be better to check all packages, at upgrade the first package might be the old one from the system while the new data is present in the new package on the medium...